File: packages/go/contracts/commands/indicator.go
package commands

import (
	"time"

	"trend-buddy/packages/go/contracts/domain"
)

type IndicatorRequest struct {
	Indicator    string              `json:"indicator"`
	InstrumentID domain.InstrumentID `json:"instrument_id"`
	Timeframe    domain.Timeframe    `json:"timeframe"`
	StartTime    time.Time           `json:"start_time"`
	EndTime      time.Time           `json:"end_time"`
	Parameters   map[string]any      `json:"parameters,omitempty"`
}

File: packages/python/src/trend_buddy_shared/contracts/commands/indicator.py
from dataclasses import dataclass, field
from datetime import datetime
from typing import Any

from trend_buddy_shared.contracts.domain.instrument import Timeframe


@dataclass
class IndicatorRequest:
    indicator: str
    instrument_id: str
    timeframe: Timeframe
    start_time: datetime
    end_time: datetime
    parameters: dict[str, Any] = field(default_factory=dict)
